MouseUnSnag.cs

```diff
@@ -437,7 +438,7 @@ bool CheckJumpCursor (Point mouse, Point cursor, out Point NewCursor)
         {
             NewCursor = jumpScreen.R.ClosestBoundaryPoint (cursor);
         }
-        else if (StuckDirection.X != 0)
+        else if (wrap && StuckDirection.X != 0)

```

Comment:
... and then negate that here:

```suggestion
        else if (!nowrap && StuckDirection.X != 0)
```